mythril/analysis/call_helpers.py

```diff
@@ -0,0 +1,49 @@
+from mythril.analysis.ops import VarType, Call, get_variable
+
+
+def get_call_from_state(state):
+    instruction = state.get_current_instruction()
+
+    op = instruction["opcode"]
+
+    stack = state.mstate.stack
+
+    if op in ("CALL", "CALLCODE"):
+        gas, to, value, meminstart, meminsz, memoutstart, memoutsz = (
+            get_variable(stack[-1]),
+            get_variable(stack[-2]),
+            get_variable(stack[-3]),
+            get_variable(stack[-4]),
+            get_variable(stack[-5]),
+            get_variable(stack[-6]),
+            get_variable(stack[-7]),
+        )
+
+        if to.type == VarType.CONCRETE and to.val < 5:
+            return
+
+        if meminstart.type == VarType.CONCRETE and meminsz.type == VarType.CONCRETE:
+            return Call(
+                state.node,
+                state,
+                None,
+                op,
+                to,
+                gas,
+                value,
+                state.mstate.memory[meminstart.val : meminsz.val * 4],
+            )
+        else:
+            return Call(state.node, state, None, op, to, gas, value)
+
+    else:
+        gas, to, meminstart, meminsz, memoutstart, memoutsz = (
+            get_variable(stack[-1]),
+            get_variable(stack[-2]),
+            get_variable(stack[-3]),
+            get_variable(stack[-4]),
+            get_variable(stack[-5]),
+            get_variable(stack[-6]),
+        )
+
+        return Call(state.node, state, None, op, to, gas)
```

Comment:
Do we want to check if `meminstart` and `meminsz` are concrete here and potentially create the Call with the calldata from memory?

Reply:
I don't think we currently need it, but do we need/use it in the above case either? There might be a case where it's useful to extract the calldata for a delegatecall or different opcode, but it's not currently needed. You make the call here (that was also a pun btw).

Reply:
Just to clarify this `Call` is the class in the `mythril.analysis.ops` which comes handy to manage Call type opcodes right?

Reply:
That's what I understood.

Reply:
It's easy to maintain various variables with call, so lets leave it that way, WDYT?

Reply:
That's fine. :ok_hand: